cmd/tmsUpload.go

```diff
@@ -186,103 +152,58 @@ func runTmsUpload(config tmsUploadOptions, communicationInstance tms.Communicati
 			}
 		}
 	}
+	return nil
+}
 
-	fileInfo, errUploadFile := communicationInstance.UploadFile(mtaPath, namedUser)
-	if errUploadFile != nil {
-		log.SetErrorCategory(log.ErrorService)
-		return fmt.Errorf("failed to upload file: %w", errUploadFile)
+func tmsUploadFile(config tmsUploadOptions, communicationInstance tms.CommunicationInterface, utils tms.TmsUtils) (string, error) {
+	mtaPath := config.MtaPath
+	exists, _ := utils.FileExists(mtaPath)
+	if !exists {
+		log.SetErrorCategory(log.ErrorConfiguration)
+		return "", fmt.Errorf("mta file %s not found", mtaPath)
 	}
 
-	_, errUploadFileToNode := communicationInstance.UploadFileToNode(nodeName, strconv.FormatInt(fileInfo.Id, 10), description, namedUser)
-	if errUploadFileToNode != nil {
+	fileInfo, errUploadFile := communicationInstance.UploadFile(mtaPath, config.NamedUser)
+	if errUploadFile != nil {
 		log.SetErrorCategory(log.ErrorService)
-		return fmt.Errorf("failed to upload file to node: %w", errUploadFileToNode)
+		return "", fmt.Errorf("failed to upload file: %w", errUploadFile)
 	}
 
-	return nil
+	fileId := strconv.FormatInt(fileInfo.Id, 10)
+	return fileId, nil
 }
 
-func formNodeIdExtDescriptorMappingWithValidation(utils tmsUploadUtils, nodeNameExtDescriptorMapping map[string]interface{}, nodes []tms.Node, mtaYamlMap map[string]interface{}, mtaVersion string) (map[int64]string, error) {
-	var wrongMtaIdExtDescriptors []string
-	var wrongExtDescriptorPaths []string
-	var wrongNodeNames []string
-	var errorMessage string
-
-	nodeIdExtDescriptorMapping := make(map[int64]string)
-	for nodeName, mappedValue := range nodeNameExtDescriptorMapping {
-		mappedValueString := fmt.Sprintf("%v", mappedValue)
-		exists, _ := utils.FileExists(mappedValueString)
-		if exists {
-			extDescriptorMap, errGetYamlAsMap := getYamlAsMap(utils, mappedValueString)
-			if errGetYamlAsMap == nil {
-				if fmt.Sprintf("%v", mtaYamlMap["ID"]) != fmt.Sprintf("%v", extDescriptorMap["extends"]) {
-					wrongMtaIdExtDescriptors = append(wrongMtaIdExtDescriptors, mappedValueString)
-				}
-			} else {
-				wrappedErr := errors.Wrapf(errGetYamlAsMap, "tried to parse %v as yaml, but got an error", mappedValueString)
-				errorMessage += fmt.Sprintf("%v\n", wrappedErr)
-			}
-		} else {
-			wrongExtDescriptorPaths = append(wrongExtDescriptorPaths, mappedValueString)
-		}
-
-		isNodeFound := false
-		for _, node := range nodes {
-			if node.Name == nodeName {
-				nodeIdExtDescriptorMapping[node.Id] = mappedValueString
-				isNodeFound = true
-				break
-			}
-		}
-		if !isNodeFound {
-			wrongNodeNames = append(wrongNodeNames, nodeName)
-		}
-	}
+func tmsUpload(config tmsUploadOptions, telemetryData *telemetry.CustomData, influx *tmsUploadInflux) {
```

Comment:
This step could be as "empty" as the tmsExport step. All functions used by both steps should be moved to the pkg IMO